### ISSUE TITLE
Add Crystal language support

### ISSUE THREAD

**Issue by spindlebink** [helix-editor/helix#1679]
The built-in Ruby syntax works fine for Crystal files (I added a `.cr` file type to `languages.toml`) but there's already a [Crystal tree-sitter on GitHub](https://github.com/keidax/tree-sitter-crystal) that I don't imagine would be too hard to add, providing somebody can write query files for it.

**Comment by spindlebink** [comment]
Alright, have taken a shot at writing the query for this, since Ruby's parser borks at the floored division operator `//`, and at a second look it seems that the linked implementation might not be complete. I know next to nothing about tree sitter implementations, though, so I could be mistaken--will probably need someone with expertise to check or possibly the author of the original repo @keidax to chime in.

**Comment by keidax** [comment]
That's correct, [tree-sitter-crystal](https://github.com/keidax/tree-sitter-crystal) is very much a work in progress. I'll add a README notice to make that clearer.

**Comment by kpbaks** [comment]
This issue can be closed now. Since c9664212 we have support for Crystal's LSP, and all the tree-sitter query categories.

![Image](https://github.com/user-attachments/assets/3f0d3dd2-b6d1-4bd2-9961-bb38eb29d3a6)
